src/execution/operator/helper/physical_limit.cpp

```diff
@@ -13,28 +15,61 @@ class PhysicalLimitOperatorState : public PhysicalOperatorState {
 	idx_t current_offset;
 };
 
+uint64_t get_delimiter(DataChunk &input, Expression *expr) {
+	DataChunk limit_chunk;
+	vector<LogicalType> types{expr->return_type};
+	limit_chunk.Initialize(types);
+	ExpressionExecutor limit_executor(expr);
+	if (expr->type == ExpressionType::BOUND_FUNCTION) {
+		limit_executor.Execute(limit_chunk);
+	} else {
+		limit_executor.Execute(input, limit_chunk);
+	}
+	auto limit_value = limit_chunk.GetValue(0, 0);
+	if ((limit_chunk.data[0].vector_type != VectorType::CONSTANT_VECTOR && limit_chunk.size() != 1) || limit_chunk.ColumnCount() != 1){
+		throw Exception("Value for Limit returns number of rows/columns different than 1");
+	}
+	bool success_cast = limit_value.TryCastAs(LogicalType::UBIGINT);
+	if (success_cast) {
+		return limit_value.value_.ubigint;
+	} else {
+		throw Exception("Limit expression can't be converted to unsigned integer");
+	}
+}
+
 void PhysicalLimit::GetChunkInternal(ExecutionContext &context, DataChunk &chunk, PhysicalOperatorState *state_) {
 	auto state = reinterpret_cast<PhysicalLimitOperatorState *>(state_);
 
 	idx_t max_element = limit + offset;
-	if (limit == 0 || state->current_offset >= max_element) {
+	if ((limit == 0 || state->current_offset >= max_element) && !(limit_expression || offset_expression)) {
 		return;
 	}
 
 	// get the next chunk from the child
 	do {
 		children[0]->GetChunk(context, state->child_chunk, state->child_state.get());
+		if (limit_expression) {
+			limit = get_delimiter(state->child_chunk,limit_expression.get());
+			limit_expression.reset();
+		}
+		if (offset_expression) {
+			offset = get_delimiter(state->child_chunk,offset_expression.get());
+			offset_expression.reset();
+		}
+		max_element = limit + offset;
 		if (state->child_chunk.size() == 0) {
 			return;
 		}
-
+		if ((limit == 0 || state->current_offset >= max_element) && !(limit_expression || offset_expression)) {
```

Comment:
Here limit_expression and offset_expression should always be `nullptr`, is the second part of this check required?